[jinnsuke] iP

jinnsuke opened the pull request:
# ShoAI
> "The only way to do great work is to love what you do.” – Steve Jobs

ShoAI is designed to help users streamline their workflow with AI-powered solutions. It's:

- User-friendly
- Efficient
- **Highly customizable**

All you need to do is:

1. Clone the repository.
2. Build the project.
3. Start using ShoAI to enhance your productivity. 😎

Features:
- [x] Core AI functionalities
- [ ] Advanced analytics (coming soon)
- [ ] Customizable settings (coming soon)

If you're a Java developer, you can also use it to sharpen your Java skills. Here's a sample code snippet from the project:

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Welcome to ShoAI!");
    }
}
```

*You can find more information [here](https://github.com/jinnsuke/ip).*

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jinnsuke committed:
Rename, Greet, Exit

jinnsuke committed:
Echo

jinnsuke committed:
Add, List

jinnsuke committed:
Mark as Done

jinnsuke committed:
ToDo, Event, Deadline

jinnsuke committed:
A-TextUiTesting & Removed Indentations

jinnsuke committed:
Revert "A-TextUiTesting & Removed Indentations"

This reverts commit 671075826535efe58371b545ebdb276275ab19d5.

jinnsuke committed:
Revert "Revert "A-TextUiTesting & Removed Indentations""

This reverts commit c442918d1266ddbd3f02c13dede3d2e1b73ebf13.

jinnsuke committed:
Handle Errors

jinnsuke committed:
Delete

jinnsuke committed:
Save

jinnsuke committed:
Merge branch 'branch-Level-7'

jinnsuke committed:
Use More OOP

jinnsuke committed:
Merge tag 'A-MoreOOP'

jinnsuke committed:
Organise into Packages

jinnsuke committed:
Merge tag 'A-Packages'

jinnsuke committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Use Gradle

jinnsuke committed:
Add ShoAI.txt to gitignore

jinnsuke committed:
Add JUnit Tests

jinnsuke committed:
Merge branch 'A-JUnit'

jinnsuke committed:
Dates and Times

jinnsuke committed:
Merge tag 'Level-8'

jinnsuke committed:
Create a JAR File

jinnsuke committed:
Merge tag 'A-Jar'

jinnsuke committed:
JavaDoc

jinnsuke committed:
Merge branch 'branch-A-JavaDoc'

jinnsuke committed:
Follow the Coding Standard

jinnsuke committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/shoai/Todo.java
#	src/main/java/shoai/Ui.java

jinnsuke committed:
Find

jinnsuke committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/shoai/Parser.java

jinnsuke committed:
Configured Checkstyle

jinnsuke committed:
Merge tag 'A-CheckStyle'

Green-Tea-123 commented on src/main/java/shoai/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+ * a deadline date.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
Perhaps the name of this attribute can be more descriptive? e.g. `byDate`

Green-Tea-123 commented on src/main/java/shoai/Event.java:
```diff
@@ -0,0 +1,53 @@
+ * a start and end date for the event.
+ */
+public class Event extends Task {
+    protected LocalDate from;
+    protected LocalDate to;
```
Perhaps these two attribute can be more descriptive as well?

Green-Tea-123 commented on src/main/java/shoai/Parser.java:
```diff
@@ -0,0 +1,197 @@
+     * @return true if the command is "bye" and the application should exit, false otherwise.
+     * @throws ShoAIException If there is an error processing the command.
+     */
+    public boolean parse(String fullCommand, TaskList tasks, Ui ui, Storage storage) throws ShoAIException {
```
Perhaps a name more descriptive of the function contents should be used, given that this function does more than just parsing the argument. Maybe something like `processCommand` would be better?

Green-Tea-123 reviewed (commented):
Overall, the code quality, in terms of naming, is good.

jinnsuke committed:
Add Graphical User Interface

jinnsuke committed:
Merge tag 'GUI'

jinnsuke committed:
Add tag properly

jinnsuke committed:
Merge tag 'Level-10'

jialegitgud commented on src/main/java/shoai/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+package shoai;
+import java.time.LocalDate;
```
Imports do not seem to be in lexicographical order, perhaps the order of the imports could be changed for better readibility?

jialegitgud commented on src/main/java/shoai/Parser.java:
```diff
@@ -0,0 +1,214 @@
+    public String parse(String fullCommand, TaskList tasks, Storage storage) throws ShoAIException {
+        String[] words = fullCommand.split(" ", 2);
+        String command = words[0];
+        StringBuilder response = new StringBuilder();
```
I like the use of StringBuilder here, looks like an apt choice 👍

jialegitgud commented on src/main/java/shoai/ShoAI.java:
```diff
@@ -0,0 +1,71 @@
+    private Storage storage;
+    private Parser parser;
+    private Ui ui; // Add an instance of Ui
+    private boolean welcomed; // Flag to check if welcome message has been shown
```
Boolean attribute could perhaps be renamed to reflect it more as a boolean. eg. `isWelcomed`

jialegitgud commented on src/main/java/shoai/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+     * Returns the task at the specified index.
+     *
+     * @param index The index of the task to retrieve.
+     * @return The task at the specified index.
```
Just a small nit, maybe `@throws` could be added into the javadoc to increase clarity.

jialegitgud reviewed (commented):
In general, code looks really clean and good to me! I think there is very good effort in comprehensive documentation with JavaDocs.

jinnsuke committed:
Updated UI layout and fixed the lack of welcome message

jinnsuke committed:
Enhance Event and Deadline Classes to Handle Date and Time

jinnsuke committed:
Fix JavaFX module path configuration in Gradle build script

Corrected the configuration for JavaFX module paths in the `build.gradle` file. This fix addresses issues with running the application due to incorrect module paths and missing dependencies.

The changes include:
- Updated the `run` task to properly set the module path for JavaFX controls and FXML.
- Ensured that the `tasks.withType(JavaExec)` configuration includes all necessary JavaFX modules.
- Verified that JavaFX modules are included in the runtime classpath to avoid runtime errors.

jinnsuke committed:
Add assertions to all classes to enforce key assumptions

- Added assertions across all classes to enforce critical assumptions and validate method inputs.
- Improved code robustness by validating preconditions and invariants with assertions.

jinnsuke committed:
Refactor Parser class to improve readability and maintainability

- Extracted command handling logic into separate methods for better clarity and adherence to the Single Responsibility Principle.
- Consolidated error handling and validation into reusable methods to reduce code duplication.

jinnsuke committed:
Merge pull request #1 from jinnsuke/branch-A-Assertions

Add assertions to all classes to enforce key assumptions

jinnsuke committed:
Change Deadline Test to coincide with new Deadline class

jinnsuke committed:
Merge branch 'master' of https://github.com/jinnsuke/ip

# Conflicts:
#	src/test/java/shoai/DeadlineTest.java

jinnsuke committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/shoai/Parser.java

jinnsuke committed:
Merge pull request #2 from jinnsuke/branch-A-CodeQuality

Refactor Parser class to improve readability and maintainability

jinnsuke committed:
Merge branch 'master' of https://github.com/jinnsuke/ip

jinnsuke committed:
Set up Continuous Integration (CI)

jinnsuke committed:
Add client management functionality

- Added functionality to create, read, update, and delete client records.
- Integrated client data with the existing system to enable seamless management.

jinnsuke committed:
Merge branch 'D-Clients'

jinnsuke committed:
Enhance GUI for Asymmetric Conversation and Error Highlighting
- Implemented distinct visual styles for user inputs and app responses.
- Added error highlighting to ensure commands and errors are easily distinguishable and attention-catching.

jinnsuke committed:
Merge tag 'A-BetterGui'

jinnsuke committed:
Add a playful personality to ShoAI's messages

jinnsuke committed:
Merge tag 'A-Personality'

jinnsuke committed:
Deal with trailing/leading spaces and special characters

jinnsuke committed:
Merge tag 'A-MoreErrorHandling'

jinnsuke committed:
Set Title Header to GUI

jinnsuke committed:
Update README.md and add Ui.png

jinnsuke committed:
Update image path

jinnsuke committed:
Add Javadoc comments

jinnsuke committed:
Remove emojis and fix build.gradle file

jinnsuke committed:
Create MainApp to reroute logic flow

jinnsuke committed:
Make the DialogBox and Window resizable
